paddle/fluid/imperative/layer.cc

```diff
@@ -185,12 +185,12 @@ class Autograd {
         for (const auto& map : candidate->grad_output_vars_) {
           for (const auto& it : map) {
             for (const auto& vb : it.second) {
-              if (grad_ref->find(vb) == grad_ref->end()) {
-                grad_ref->insert(std::make_pair(vb, 1));
+              if (grad_ref.find(vb) == grad_ref.end()) {
+                grad_ref.insert(std::make_pair(vb, 1));
```

Comment:
C++ guarantees zero-initialization of primitive types. See [here](https://en.cppreference.com/w/cpp/language/zero_initialization). You can just write `++grad_ref[vb];`